alexandrie/src/utils/request_log.rs

```diff
@@ -22,17 +22,8 @@ impl<State: Clone + Send + Sync + 'static> Middleware<State> for RequestLogger {
         let res = next.run(req).await;
         let elapsed = start.elapsed().as_millis();
         let status = res.status();
-        let resp_msg = {
-            if let Some(err) = res.error() {
-                err.to_string()
-            } else {
-                "OK".into()
-            }
-        };
-        info!(
-            "--> {} {} {} {}ms , {}",
-            method, path, status, elapsed, resp_msg
-        );
+        let msg = res.error().map(|e| e.to_string()).unwrap_or("OK".into());
```

Comment:
Because of the use of `unwrap_or`, its argument (`"OK".into()`) gets evaluated unconditionally, even if its value isn't needed.  

This isn't a problem for small trivially-copyable types, like integral types, but since `"OK".into()` creates a `String` from an `&'static str`, evaluating it causes an heap-allocation and therefore is not very cheap to perform everytime.  

We can avoid this problem by using `unwrap_or_else` which, because it uses a closure, will evaluate `"OK".into()` lazily.  
```suggestion
        let msg = res.error().map(|err| err.to_string()).unwrap_or_else(|| "OK".to_string());
```

Reply:
That's great.Has fixed